### PR TITLE
Refactor batch system

### PR DESCRIPTION
**Pull request by coocood**
The old batch system for multi-thread raft is too complex.
Replace it with a simpler multi-thread model.

### CODE REVIEW

**Comment by coocood** [comment]
Each peer can bind to a worker, and we can bind a peer to another worker if we need to.

Currently, there is no scheduler to rebalance worker, simpler use the region id to bind a worker.

File: tikv/raftstore/config.go [hunk 1]
```diff
@@ -193,7 +193,7 @@ func NewDefaultConfig() *Config {
 		ApplyMaxBatchSize:        1024,
 		ApplyPoolSize:            2,
 		StoreMaxBatchSize:        1024,
-		StorePoolSize:            2,
+		StorePoolSize:            1,
```
Comment (ngaut) [inline tikv/raftstore/config.go]: Why change the default value?

Comment (coocood) [reply tikv/raftstore/config.go]: For benchmark comparison, forgot to change it back, I will make it configurable.

File: tikv/raftstore/peer_state.go [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+		storeFsm:      storeFsm,
+	}
+	for i := 0; i < workerSize; i++ {
+		pm.workerSenders[i] = make(chan Msg, 4096)
+	}
+	return pm
+}
+
+func (pr *peerRouter) get(regionID uint64) *peerState {
+	pr.mu.RLock()
```
[the remark this answers is not in the document]
Comment (coocood) [reply tikv/raftstore/peer_state.go]: We can shard the map for lower contention based on the region ID later.

[the remark this answers is not in the document]
Comment (bobotu) [reply tikv/raftstore/peer_state.go]: `sync.Map` or some kinds of COW map may also good

**Pull request merged**